packages/realm/src/schema/normalize_property_schema_pseudo_code.ts

```diff
@@ -0,0 +1,100 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+// --------------------------------------------------------------------------------------------------------------------------
```

Comment:
Good idea, I mainly created them as starting points for writing the parser. I could probably just delete the pseudo code file but add the grammar part into `normalize.ts` like you said?

Reply:
I think it's good to keep such notes around for reference later.  Normally this would be defined in a Tech Design, but I think it's nice to have it also out in the open.  The pseudo code could probably be removed, now that we have actual code ;). The grammar is useful, so that the intention of the code is clear.